src/gui/src/scriptWidget.cpp

```diff
@@ -385,8 +414,17 @@ class ScriptWidget::GuiSink : public spdlog::sinks::base_sink<Mutex>
         std::string(formatted.data(), formatted.size()));
 
     if (msg.level == spdlog::level::level_enum::off) {
-      // this comes from a ->report
-      widget_->addReportToOutput(formatted_msg);
+      if (QThread::currentThread() == widget_->thread()) {
+        if (!widget_->report_timer_->isActive()) {
+          widget_->startReportTimer();
+        }
+
+        widget_->addMsgToReportBuffer(formatted_msg);
```

Comment:
you still need to add the message, you just can't trigger the timer or the event loop. Just move this line up to before the thread check.